runtime/autoload/provider/clipboard.vim

```diff
@@ -46,48 +46,106 @@ function! provider#clipboard#Error() abort
   return s:err
 endfunction
 
+let s:providers = [
+      \ {
+      \   'name': 'pbcopy/pbpaste',
+      \   'copy': {
+      \      '+': 'pbcopy',
+      \      '*': 'pbcopy',
+      \    },
+      \   'paste': {
+      \      '+': 'pbpaste',
+      \      '*': 'pbpaste',
+      \   },
+      \   'cache_enabled': 0,
+      \   'check': "executable('pbcopy')",
```

Comment:
I explicitly removed the `has('mac')` check from here because now that it is configurable, we don't need to make assumptions.